flow/Arena.h

```diff
@@ -91,22 +91,23 @@ class NonCopyable
 
 class Arena {
```

Comment:
This is my first time looking at Arena, and a high-level description would help.  I think this is right.  Could you add something like this (but hopefully correct) here:

> An Arena is a custom allocator that consists of a set of ArenaBlocks.  Allocation is performed by bumping a pointer on the most recent ArenaBlock until the block is unable to service the next allocation request.  When the current ArenaBlock is full, a new (larger) one is added to the Arena.  Deallocation is not directly supported.  Instead, memory is freed by deleting the entire Arena at once.

Reply:
Done